library-best-practices/JLBP-17.md

```diff
@@ -0,0 +1,44 @@
+[JLBP-17] Coordinate Rollout of Breaking Changes
+------------------------------------------------
+
+When a library in a dependency tree introduces a breaking change, consumers
+can't update to that version until all their other dependencies which also use
+the library update to that version first. To enable clients to depend on the
+latest versions of all their dependencies, propagate breaking changes as
+quickly as possible.
+
+Perform the rollout in this manner:
+
+1. Decide whether to introduce the incompatibility in a single release or
+   make the change in two phases.
+   - The two-phase approach is described in [JLBP-7: Make breaking transitions
+     easy](JLBP-7.md).
+   - If the feature is used by stable code in other libraries, you must use the
+     two-phase approach, except for the types of changes where this is
+     impossible (see below).
+   - If the code does not appear to be used in other libraries, an in-place
+     breakage may be ok.
+   - If the feature is used by unstable code or stable code having little usage
+     itself, prefer two phases if possible, with in-place breakage only if a
+     two-phase rollout is judged to be too costly.
+   - Some types of changes cannot be done using a two-phase approach, for
+     example, making classes or methods `final` or adding non-defaulted methods
```

Comment:
non-defaulted methods --> methods to interfaces without a default implementation

Reply:
Will do.